### PR TITLE
Update subscription client with new api version

### PR DESCRIPTION
**Pull request by vivsriaus**

### CODE REVIEW

**Comment by TianoMS** [comment]
LGTM.

File: arm-resources/subscriptions/2016-06-01/swagger/subscriptions.json [hunk 1]
```diff
@@ -0,0 +1,312 @@
...
+    "title": "SubscriptionClient",
+    "version": "2016-06-01"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json",
+    "text/json"
```
Comment (amarzavery) [inline arm-resources/subscriptions/2016-06-01/swagger/subscriptions.json]: does the service actually support text/json in produces and consumes? When we tested this aspect for one of the services, it returned a 400 bad request with "text/json". So we wanted to confirm.

File: arm-resources/subscriptions/2016-06-01/swagger/subscriptions.json [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+            "type": "string",
+            "description": "Id of the subscription"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "",
```
Comment (amarzavery) [inline arm-resources/subscriptions/2016-06-01/swagger/subscriptions.json]: please make sure that description is not empty. Empty description generates bad client library documentation.

File: arm-resources/subscriptions/2016-06-01/swagger/subscriptions.json [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+            "type": "string",
+            "description": "Id of the subscription."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "",
```
Comment (amarzavery) [inline arm-resources/subscriptions/2016-06-01/swagger/subscriptions.json]: no empty description: general comment

File: arm-resources/subscriptions/2016-06-01/swagger/subscriptions.json [hunk 4]
```diff
@@ -0,0 +1,312 @@
...
+        }
+      }
+    },
+    "/subscriptions": {
+      "get": {
+        "tags": [
+          "Subscriptions"
+        ],
+        "operationId": "Subscriptions_List",
+        "description": "Gets a list of the subscriptionIds.",
```
Comment (amarzavery) [inline arm-resources/subscriptions/2016-06-01/swagger/subscriptions.json]: "Gets a list of subscriptions." As the operation returns a list of subscription objects and not just ids.

File: arm-resources/subscriptions/2016-06-01/swagger/subscriptions.json [hunk 5]
```diff
@@ -0,0 +1,312 @@
...
+        }
+      }
+    },
+    "/tenants": {
+      "get": {
+        "tags": [
+          "Tenants"
+        ],
+        "operationId": "Tenants_List",
+        "description": "Gets a list of the tenantIds.",
```
Comment (amarzavery) [inline arm-resources/subscriptions/2016-06-01/swagger/subscriptions.json]: "Gets a list of tenants."

File: arm-resources/subscriptions/2016-06-01/swagger/subscriptions.json [hunk 6]
```diff
@@ -0,0 +1,312 @@
...
+        "tenantId": {
+          "type": "string",
+          "description": "The tenant Id."
+        },
+        "displayName": {
+          "type": "string",
+          "description": "The subscription display name."
+        },
+        "state": {
+          "type": "string",
```
Comment (amarzavery) [inline arm-resources/subscriptions/2016-06-01/swagger/subscriptions.json]: Should this be an enum, as there will only be a finite set of values.

File: arm-resources/subscriptions/2016-06-01/swagger/subscriptions.json [hunk 7]
```diff
@@ -0,0 +1,312 @@
...
+          "description": "The authorization source of the request. Valid values are one or more combinations of Legacy, RoleBased, Bypassed, Direct and Management. For example, 'Legacy, RoleBased'."
+        }
+      },
+      "description": "Subscription information."
+    },
+    "SubscriptionPolicies": {
+      "properties": {
+        "locationPlacementId": {
+          "type": "string",
+          "description": "The subscription location placement Id."
```
Comment (amarzavery) [inline arm-resources/subscriptions/2016-06-01/swagger/subscriptions.json]: what does this mean? The description does not provide any meaningful information. Can you provide a link to the MSDN doc that provides more info?

File: arm-resources/subscriptions/2016-06-01/swagger/subscriptions.json [hunk 8]
```diff
@@ -0,0 +1,312 @@
...
+      "required": [
+        "nextLink"
+      ],
+      "description": "Subscription list operation response."
+    },
+    "TenantIdDescription": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "The Id."
```
Comment (amarzavery) [inline arm-resources/subscriptions/2016-06-01/swagger/subscriptions.json]: very confusing. what is the difference between the id  and the tenantId?

File: arm-resources/subscriptions/2016-06-01/swagger/subscriptions.json [hunk 9]
```diff
@@ -0,0 +1,312 @@
...
+          "description": "The locations."
+        }
+      },
+      "description": "Location list operation response."
+    },
+    "Subscription": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "The ID of the resource (/subscriptions/SubscriptionId)."
```
Comment (amarzavery) [inline arm-resources/subscriptions/2016-06-01/swagger/subscriptions.json]: Do you mean that subscription, location and tenant are resources? It looks really weird.

File: arm-resources/subscriptions/2016-06-01/swagger/subscriptions.json (hunk 1 above)
Comment (vivsriaus) [inline arm-resources/subscriptions/2016-06-01/swagger/subscriptions.json]: Great catch! Yep, we just support application/json. Will change this

File: arm-resources/subscriptions/2016-06-01/swagger/subscriptions.json (hunk 2 above)
Comment (vivsriaus) [inline arm-resources/subscriptions/2016-06-01/swagger/subscriptions.json]: What would be a good example text for response? I see some specs that just says "OK", some still empty, and some a brief description of what a normal response returns. Is there a standard format that you guys recommend?

File: arm-resources/subscriptions/2016-06-01/swagger/subscriptions.json (hunk 4 above)
Comment (vivsriaus) [inline arm-resources/subscriptions/2016-06-01/swagger/subscriptions.json]: Thanks

File: arm-resources/subscriptions/2016-06-01/swagger/subscriptions.json (hunk 5 above)
Comment (vivsriaus) [inline arm-resources/subscriptions/2016-06-01/swagger/subscriptions.json]: Thanks

File: arm-resources/subscriptions/2016-06-01/swagger/subscriptions.json (hunk 6 above)
Comment (vivsriaus) [inline arm-resources/subscriptions/2016-06-01/swagger/subscriptions.json]: Indeed!

File: arm-resources/subscriptions/2016-06-01/swagger/subscriptions.json (hunk 2 above)
Comment (amarzavery) [inline arm-resources/subscriptions/2016-06-01/swagger/subscriptions.json]: We usually recommend providing a meaningful description for the response. In scenarios where that is not possible, falling back on the English phrase of the status code is fine.

File: arm-resources/subscriptions/2016-06-01/swagger/subscriptions.json (hunk 7 above)
Comment (vivsriaus) [inline arm-resources/subscriptions/2016-06-01/swagger/subscriptions.json]: Will add more description

File: arm-resources/subscriptions/2016-06-01/swagger/subscriptions.json (hunk 8 above)
Comment (vivsriaus) [inline arm-resources/subscriptions/2016-06-01/swagger/subscriptions.json]: Will add an example

File: arm-resources/subscriptions/2016-06-01/swagger/subscriptions.json (hunk 9 above)
Comment (vivsriaus) [inline arm-resources/subscriptions/2016-06-01/swagger/subscriptions.json]: Technically, we call them the fully qualified resource id, but I can see how that could be confusing. Will remove the word 'resource' and add an example to disambiguate

File: arm-resources/subscriptions/2016-06-01/swagger/subscriptions.json (hunk 3 above)
Comment (vivsriaus) [inline arm-resources/subscriptions/2016-06-01/swagger/subscriptions.json]: Addressed

File: arm-resources/subscriptions/2016-06-01/swagger/subscriptions.json (hunk 2 above)
Comment (vivsriaus) [inline arm-resources/subscriptions/2016-06-01/swagger/subscriptions.json]: Sounds good.

File: arm-resources/subscriptions/2016-06-01/swagger/subscriptions.json [hunk 10]
```diff
@@ -0,0 +1,321 @@
...
+          },
+          "description": "The locations."
+        }
+      },
+      "description": "Location list operation response."
+    },
+    "Subscription": {
+      "properties": {
+        "id": {
+          "type": "string",
```
Comment (johanste) [inline arm-resources/subscriptions/2016-06-01/swagger/subscriptions.json]: Shouldn't id be readonly?

Comment (vivsriaus) [inline arm-resources/subscriptions/2016-06-01/swagger/subscriptions.json]: All the properties  of location, subscription and tenant are readonly indeed. We only have GET apis for these. Updated with my last commit

**Pull request merged**